.github/workflows/docs.yml

```diff
@@ -0,0 +1,49 @@
+name: Build and deploy Documentation site
+
+on:
+  push:
+    branches:
+      - master
+
+env:
+  RUBY_VERSION: 2.7
+  
+  # For control over build environment
+  DOCS_DEPLOY: true
+
+jobs:
+  deploy_docs:
+    runs-on: "ubuntu-latest"
+    env:
+      BUNDLE_PATH: "vendor/bundle"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ env.RUBY_VERSION }}
+          bundler-cache: true
+      - name: Clone target branch
+        run: |
+          REMOTE_BRANCH="${REMOTE_BRANCH:-gh-pages}"
+          REMOTE_REPO="https://${GITHUB_ACTOR}:${{ secrets.GITHUB_TOKEN }}@github.com/${GITHUB_REPOSITORY}.git"
+
+          echo "Publishing to ${GITHUB_REPOSITORY} on branch ${REMOTE_BRANCH}"
+          rm -rf docs/_site/
+          git clone --depth=1 --branch="${REMOTE_BRANCH}" --single-branch --no-checkout \
+            "${REMOTE_REPO}" docs/_site/
+      - name: Build site
+        run: bundle exec jekyll build --source docs --destination docs/_site --verbose --trace
+      - name: Deploy to GitHub Pages
+        run: |
+          SOURCE_COMMIT="$(git log -1 --pretty="%an: %B" "$GITHUB_SHA")"
+          pushd docs/_site &>/dev/null
+          : > .nojekyll
```

Comment:
Why not use "touch" here? This command is a bit obtuse.

Reply:
`touch <file>` and `: > <file>` is subtly different even though they appear to do the same at first glance.

`touch <file>` creates `<file>` if it doesn't exist and just modifies *last accessed time `atime`* on subsequent uses.

`: > <file>` also creates `<file>` if it doesn't exist but will *truncate its length to `0`* (changes `mtime`) on subsequent uses.

Ref: [Stack Exchange Thread](https://askubuntu.com/questions/831414/what-is-the-difference-between-touch-file-and-file).

---

For argument's sake, one could state that in the current context, the maintainers have total control over `.nojekyll` never being a *non-empty* file and that its contents are irrelevant regardless.